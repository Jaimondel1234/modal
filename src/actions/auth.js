import { types } from "../types/types";

export const authLogin = (email, password) => ({
  type: types.authLogin,
  email,
  password,
});

export const authLogout = () => ({
  type: types.authLogout,
});
