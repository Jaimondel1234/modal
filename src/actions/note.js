import { types } from "../types/types";

export const noteAddNew = (newNotes) => ({
  type: types.noteAddNew,
  newNotes,
});

export const noteDelete = (noteId) => ({
  type: types.noteDeleted,
  noteId,
});

export const noteSetActive = (activeNote) => ({
  type: types.noteSetActive,
  activeNote,
});

export const noteUpdate = (note) => ({
  type: types.noteUpdate,
  note,
});

export const setListMode = (mode) => ({
  type: types.noteSetListMode,
  mode,
});

export const noteReset = () => ({
  type: types.noteReset,
});
