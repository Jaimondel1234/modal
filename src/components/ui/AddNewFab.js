import React from "react";
import { faPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import styled from "styled-components";
import { useDispatch } from "react-redux";
import { uiOpenModal } from "../../actions/ui";
import { FabStyle } from "../../styles/core";

const PlusFabStyle = styled(FabStyle)`
  border-radius: 100%;
  font-size: 50px;
  padding: 30px;
  position: fixed;
  right: 40px;
  bottom: 25px;
  color: white;
  background-color: #007bff;
`;

/*
  Button to Open the modal
*/
export const AddNewFab = () => {
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(uiOpenModal());
  };
  return (
    <PlusFabStyle onClick={handleClick}>
      <FontAwesomeIcon icon={faPlus} />
    </PlusFabStyle>
  );
};
