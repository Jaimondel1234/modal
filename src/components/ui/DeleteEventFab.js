import React from "react";
import PropTypes from "prop-types";
import { faTrash } from "@fortawesome/free-solid-svg-icons/faTrash";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { useDispatch } from "react-redux";
import styled from "styled-components";
import { noteDelete } from "../../actions/note";
import { FabStyle } from "../../styles/core";

const TrashFabStyle = styled(FabStyle)`
  left: 8%;
  bottom: 75%;
`;

/*
  Button to delete a note
*/
export const DeleteEventFab = ({ noteId }) => {
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(noteDelete(noteId));
  };

  return (
    <TrashFabStyle onClick={handleClick}>
      <FontAwesomeIcon icon={faTrash} />
    </TrashFabStyle>
  );
};

DeleteEventFab.propTypes = {
  noteId: PropTypes.number.isRequired,
};
