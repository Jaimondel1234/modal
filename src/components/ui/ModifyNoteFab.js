import React from 'react';
import PropTypes from 'prop-types';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons/faPencilAlt';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { noteSetActive } from '../../actions/note';
import { uiOpenModal } from '../../actions/ui';
import { FabStyle } from '../../styles/core';

const PencilAltFabStyle = styled(FabStyle)`
  right: 8%;
  bottom: 75%;
`;

/*
  Button to modify a note
*/
export const ModifyNoteFab = ({ note }) => {
  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(noteSetActive(note));
    dispatch(uiOpenModal());
  };

  return (
    <PencilAltFabStyle onClick={handleClick}>
      <FontAwesomeIcon icon={faPencilAlt} />
    </PencilAltFabStyle>
  );
};

ModifyNoteFab.propTypes = {
  note: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    imgUrl: PropTypes.string,
    description: PropTypes.string,
  }),
};
