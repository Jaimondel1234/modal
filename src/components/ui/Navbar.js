import React from 'react';
import { Link } from 'react-router-dom';
import { NavbarList, NavbarElement, ButtonStyle } from '../../styles/core';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';
import { noteReset } from '../../actions/note';
import { authLogout } from '../../actions/auth';

const ButtonLogoutStyle = styled(ButtonStyle)`
  display: block;
  font-weight: bold;
  font-size: 40px;
  text-align: center;
  letter-spacing: 5px;
  line-height: 60px;
`;

/* 
   Navbar
 */

export const Navbar = () => {
  const dispatch = useDispatch();

  const handleLogout = () => {
    localStorage.removeItem('user');
    localStorage.removeItem('notes');
    dispatch(authLogout());
    dispatch(noteReset());
  };
  return (
    <>
      <NavbarList>
        <NavbarElement>
          <Link to='/home'>Home</Link>
        </NavbarElement>
        <NavbarElement>
          <Link to='/notes'>Notes</Link>
        </NavbarElement>
        <NavbarElement float='right'>
          <ButtonLogoutStyle onClick={handleLogout} m='5px' p='5px 10px'>
            Logout
          </ButtonLogoutStyle>
        </NavbarElement>
      </NavbarList>
    </>
  );
};
