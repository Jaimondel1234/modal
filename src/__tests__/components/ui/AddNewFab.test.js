import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { AddNewFab } from "../../../components/ui/AddNewFab";
import { uiOpenModal } from "../../../actions/ui";

describe("Test <AddNewFab/>", () => {
  const mockStore = configureStore([]);
  const initState = {};

  const store = mockStore(initState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <AddNewFab />
    </Provider>
  );
  test("it must be displayed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("if the button is clicked, the open modal action must be dispatched", () => {
    wrapper.find("button").prop("onClick")();
    expect(store.dispatch).toHaveBeenCalledWith(uiOpenModal());
  });
});
