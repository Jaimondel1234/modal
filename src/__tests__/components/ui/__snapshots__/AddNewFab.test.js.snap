// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddNewFab/> it must be displayed correctly 1`] = `
<button
  className="core__FabStyle-sc-1mkz20q-9 AddNewFab__PlusFabStyle-sc-1cvbv4n-0 irwEJW dRoibW"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-plus fa-w-14 "
    data-icon="plus"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
